### ISSUE TITLE
Combination of backticks and tildes creates error.

### ISSUE THREAD

**Issue by sabinlinden** [tonybaloney/sphinxcontrib-confluencebuilder#62]
One of my rst files was generating this error -

```
sphinxcontrib.confluencebuilder error:
---
An unsupported Confluence API call has been made.

REQ: POST
RSP: 400
URL: https://lindenlab.atlassian.net/wiki/rest/api/
API: content
MSG: Error parsing xhtml: Unexpected close tag </code>; expected </sub>.
 at [row,col {unknown-source}]: [200,121]
---
```

When I dug in to see what it was, after a lot of trial and error I discovered the error was only triggered with a very specific combination of `` and ~ characters. The line in question contained the following string:
```
``~/.devscripts`` & ``~/.bashrc``
```

It turns out if you remove any one of the backticks or any one of the tildes, it causes the error to go away. It's this very specific combination creating the error. The ampersand doesn't seem to have any effect, either, but I wanted to include the exact string causing the bug just in case.

**Comment by jdknight** [comment]
This is an interesting bug. First look at it, it actually seems to be a Confluence parsing error. Using the markup (to reiterate):

```
``~/.devscripts`` & ``~/.bashrc``
```
While get converted by this extension to:
```
{{~/.devscripts}} & {{~/.bashrc}}
```
And with the current internal reliance to convert Wiki format to storage format, Confluence converts this section to (a detection of a code block and a subscript block):
```
<code><sub>/.devscripts</code> &amp; <code></sub>/.bashrc</code>
```
The improper HTML results in the 400 error. A user workaround I see is using the following:
```
``&#126;/.devscripts`` & ``&#126;/.bashrc``
```
Which gives a working conversion that can be published:
```
<code>&#126;/.devscripts</code> &amp; <code>&#126;/.bashrc</code>.
```
An internal workaround could be to detect the code block in the source, and if a code block's first character is a tilde, perform a conversion to the special HTML character. While it's not graceful, it might be worth it since having code blocks leading with tilde are (I would suspect be) common. Ideally, whenever an internal storage-based publisher is supported (if ever), this wouldn't be an issue.

**Comment by jdknight** [comment]
Hmm... some more investigation reveals even more problems. Replacing the tilde character with other special characters (e.x. `*`) results in the same issue. A crude way of looking at it seems like Confluence will perform any text formatting styles on a first-pass and then perform a monospace conversion on another pass.

One option could be to escape the content inside an inline block. While I feel that the inline literals shouldn't have internal text effects, I'm not 100% certain that is always desired. That being said, we can consider the following example.

Currently, this extension will take the markup:
```
``a *word* test``
```
and in Confluence, the "a word test" would render in a monospace font and the value "word" would be bolded inside. Comparing the results to an Sphinx `html` builder results in the monospace of "a \*word\* test". As this extension (I feel) keeps more inline to Sphinx documentation conventions over possibilities provided by Confluence, maybe escaping any inline literals is the right approach to addressing this issue.

**Comment by jdknight** [comment]
The issue should now be resolved with the merge of #72. The fix should be available next release (v0.9) but can also be currently used from `master`, I've marked this issue as fixed and will close.
